templates/_github/workflows/test-and-release.yml.ts

```diff
@@ -2,194 +2,117 @@ import type { TemplateFunction } from "../../../src/lib/createAdapter";
-            - "v[0-9]+.[0-9]+.[0-9]+-**"
+  push:
+    branches:
+      - "*"
+    tags:
+      # normal versions
+      - "v[0-9]+.[0-9]+.[0-9]+"
+      # pre-releases
+      - "v[0-9]+.[0-9]+.[0-9]+-**"
     pull_request: {}
 
 jobs:
-    # Performs quick checks before the expensive test runs
-    check-and-lint:
-        if: contains(github.event.head_commit.message, '[skip ci]') == false
+  # Performs quick checks before the expensive test runs
+  check-and-lint:
+    if: contains(github.event.head_commit.message, '[skip ci]') == false
 
-        runs-on: ubuntu-latest
+    runs-on: ubuntu-latest
 
-        strategy:
-            matrix:
-                node-version: [${latestNodeVersion}]
-
-        steps:
-            - name: Checkout code
-              uses: actions/checkout@v2
-
-            - name: Use Node.js \${{ matrix.node-version }}
-              uses: actions/setup-node@v2
-              with:
-                  node-version: \${{ matrix.node-version }}
-                  cache: 'npm'
-
-            - name: Install Dependencies
-              run: npm ci
-${useTypeScript ? (
-`            - name: Perform a type check
-              run: npm run build:ts -- --noEmit
-              env:
-                  CI: true`) : ""}
-${useESLint ? (
-`            - name: Lint source code
-              run: npm run lint
-`) : ""}            - name: Test package files
-              run: npm run test:package
+    steps:
+      - uses: ioBroker/testing-action-check@v1
+        with:
+          node-version: '${latestNodeVersion}'
+          # Uncomment the following line if your adapter cannot be installed using 'npm ci'
+          # install-command: 'npm install'${useTypeScript ? (`
+          check: true`) : ""}${useESLint ? (`
+          lint: true`) : ""}
 
 ${isAdapter ? (
-`    # Runs adapter tests on all supported node versions and OSes
-    adapter-tests:
-        if: contains(github.event.head_commit.message, '[skip ci]') == false
-
-        needs: [check-and-lint]
-
-        runs-on: \${{ matrix.os }}
-        strategy:
-            matrix:
-                node-version: [${adapterTestVersions.join(", ")}]
-                os: [${adapterTestOS.join(", ")}]
+`  # Runs adapter tests on all supported node versions and OSes
+  adapter-tests:
+    if: contains(github.event.head_commit.message, '[skip ci]') == false
+
+    runs-on: \${{ matrix.os }}
+    strategy:
+      matrix:
+        node-version: [${adapterTestVersions.join(", ")}]
+        os: [${adapterTestOS.join(", ")}]
 ${(adapterTestOS.includes("windows-latest") && adapterTestVersions.includes("8.x")) ? (
-	// This is unnecessary but I'm leaving it here in case we need it again.
-	// The else branch will never trigger
-`                exclude:
-                    # Don't test Node.js 8 on Windows. npm is weird here
-                    - os: windows-latest
-                      node-version: 8.x`
+  // This is unnecessary but I'm leaving it here in case we need it again.
```

Comment:
I think, we can remove this. If we ever need it again, we can simply look at the git log.

Reply:
Fine with me